### ISSUE TITLE
Improvements for the Kanji Pad

### ISSUE THREAD

**Issue by R0M4N0F5K7** [mvysny/aedict#582]
I decided to go buy the KanjiPad Extension to have all my Japanese Lookup Tools in one App and to support you, because I really like your dictionary.
I have to admit, that I strongly prefer the "Kanji Recognizer" for Kanji Drawing, as it gives me better results and for the more comfortable layout. The KanjiPad is okay but could became much more comfortable with some small tweaks:

1) MORE SPACE FOR THE DRAWING AREA
I'm using Aedict on a Xperia Z1 Compact with 4.3 inch display. When I use the KanjiPad for more complicated Kanji it feel very to squeeze them in that small area. After the Update to the paid KanjiPad Ad-on the line with the "Predict kanji strokes"-checkbox took up even more space, making the drawing area smaller. It would be very appreciated, if you could add like 2-3 cm in height to the drawing area.
Maybe you could add options in settings to set the kanji-search-result size and/or lines? and if the line number is set to just one you could make it scroll-able to left and right instead of up and down for more comfort (like the Kanji Recognizer does).

Talking about the "Predict Kanji Stroke"-Checkbox: It does remember the input and I have to check it every time again, which is quite annoying. Is their any huge disadvantage if activated anyway?
Maybe this option could be moved to the settings to - would save like 1 cm in height right away.

2) ADD A BORDER AROUND THE DRAWING AREA FOR THE BLACK THEME
This one to sounds like it's not a big deal but actually is a huge design-flaw as many phones use software android keys in form of a black bar on bottom of the screen and as they seem to use almost the same black than the drawing area on my phone it is not to clear to see where one ends and how far I can draw. In the dictionary aedict uses thin white to display the end of the screen/fields clearly without loosing much space. It would nice if you'd add that design choice to the drawing area too. Again - Kanji Recognizer can show you what I mean.

These are the main points for me. Search results are quite good in the Kanji Pad and the possibility to use the drawn kanji for further search in the dictionary with one or two clicks makes up for the slightly more accurate recognition of the Kanji Recognizer in my opinion. 

Keep up the good work! ;)

**Comment by mvysny** [comment]
Hey, thank you very much for your kind support.
1. Yeah, I know there is too much crammed in one screen :) I can slim down the check box and move it upwards, no problem - that should save some space. I can also add a configuration option for a larger kanji pad, which should change the current ratio of 33% kanji list / 66% drawing area to, say, 25% kanji list / 75% drawing area. I will implement this and paste a screenshot here, and we may discuss this further
2. Sure, no problem. I can also make the thin blue lines more visible.

**Comment by mvysny** [comment]
![screenshot](https://cloud.githubusercontent.com/assets/254091/10754039/3f8946c0-7c95-11e5-930a-1a41e3725f6f.png)

Please let me know if this is a step in the right direction.

**Comment by R0M4N0F5K7** [comment]
Yes, definitely the right direction! :)
Orientation-wise the more visible could also be beneficial, but not to sure if it will look nice on the phone. But the screenshot looks quite good.
Just the movement of the lowest line ("Kanji Pad") right on the drawing area instead of keeping it all clean and separated is rather interesting, but over all it will do the trick. Good direction indeed!

And thanks for the fast reply and modifications - I hope you implemented it just because you liked the idea and not because you try to implement every stupid idea what so ever. You obviously did a good job with aedict, don't stress yourself too much out because of minor things. ;)

**Comment by mvysny** [comment]
Hey, no problem :) I really wish that users feel Aedict controls being natural. So, if you (or anybody) has trouble drawing complex characters on the Kanji Pad, it obviously is an usability issue. If it can be remedied, great - sometimes it can't, then I suggest users to purchase a device with larger screen ;)
Fixed in Aedict 3.38.12, closing please upgrade and let me know if the kanji pad is better now.

**Comment by R0M4N0F5K7** [comment]
It does feel much better now. Even though the area is not much larger, it just feels much more intuitive and easy to draw more complex kanji in right proportions which also seems to improve the search results. More due to the writing area being more visible and squarish and less squeezed than the extra of space on it's own.
To match the kanji recognizer-apps drawing comfort, I'd would still need maybe one or half of a centimeter in height and I'd like to maybe set the KanjiPad Line Thickness to 4 (between 6 and 2 to keep small spaces and strokes distinguishable without looking to scrawly). But that's complaining on a higher level. :)
Bottom line: Great changes and small compromises - now aedict will most likely really become my all-in-one lookup app for Japanese - also for kanji. Thanks!
